### PR TITLE
Improve TorchAO quantization test coverage and XPU support

### PR DESCRIPTION
**Pull request by jiqing-feng**
## What does this PR do?

This PR improves the TorchAO quantization testing infrastructure with several fixes: enabling `int4wo` tests on Intel XPU, implementing `_dequantize` for TorchAO, fixing input dtype mismatches, and fixing training gradient underflow.

### Changes

1. **Enable int4wo tests on XPU**: Removed the `_int4wo_skip` marker that restricted `int4wo` tests to CUDA only, allowing them to run on all accelerator backends.

2. **XPU-specific int4 packing format**: Added XPU-specific handling in `_get_quant_config()` — Intel XPU requires `int4_packing_format="plain_int32"` for `Int4WeightOnlyConfig`.

3. **Fix input dtype casting**: Introduced `_get_dummy_inputs_for_model(model)` helper in `QuantizationTesterMixin` to automatically cast floating-point input tensors to the model's parameter dtype, preventing dtype mismatches during quantized model inference.

4. **Implement `_dequantize` for TorchAO**: Added `_dequantize()` method in `TorchAoHfQuantizer` that iterates all `nn.Linear` modules, calls `weight.dequantize()` on `TorchAOBaseTensor` weights, and replaces them with standard `nn.Parameter`. Also fixed `_verify_if_layer_quantized` to check `isinstance(module.weight, TorchAOBaseTensor)` so dequantized layers are correctly detected as non-quantized.

5. **Fix training gradient underflow**: Changed autocast dtype from `float16` to `bfloat16` in `_test_quantization_training`. Float16's limited dynamic range (max ~65504, min subnormal ~5.96e-8) causes gradients to underflow to zero when passing through quantized tensor subclass operations; bfloat16 shares float32's exponent range and avoids this issue.

6. **Reduce WanAnimate TorchAO test input sizes**: Shrunk dummy inputs in `TestWanAnimateTransformer3DTorchAo` to avoid OOM on devices without FlashAttention (e.g. XPU, which falls back to math SDPA and materializes the full O(S²) attention matrix). Reduced `hidden_states` from (1,36,21,64,64) to (1,36,5,16,16) and `face_pixel_values` from (1,3,77,512,512) to (1,3,13,512,512), bringing self-attention sequence length from 21,504 to 320 and peak attention memory from ~74 GiB to ~16 MB. Face frame count (13) is chosen so the face encoder's two stride-2 convolutions produce temporal output 4, plus 1 padding = 5, matching `hidden_states` temporal dim.

### CODE REVIEW

**Comment by jiqing-feng** [comment]
Hi @sayakpaul . Would you please review this PR? Thanks!

**Comment by sayakpaul** [comment]
There are a bunch of things going on in this PR. I would suggest breaking the PR into smaller PRs.

**Comment by jiqing-feng** [comment]
Hi @sayakpaul . Thanks for the review! I've split this PR into 5 smaller independent PRs as suggested:

1. **Enable TorchAO int4wo quantization tests on XPU** — [#13537](https://github.com/huggingface/diffusers/pull/13537)
   - Remove `_int4wo_skip` marker on XPU
   - Add `int4_packing_format="plain_int32"` for XPU int4 quantization

2. **Implement `_dequantize` for TorchAO quantizer** — [#13538](https://github.com/huggingface/diffusers/pull/13538)
   - Add `_dequantize()` method to `TorchAoHfQuantizer`
   - Fix `_verify_if_layer_quantized` to check `TorchAOBaseTensor` weight type

3. **Add `_get_dummy_inputs_for_model` helper** — [#13539](https://github.com/huggingface/diffusers/pull/13539)
   - Cast floating-point input tensors to model's parameter dtype automatically
   - Defined in both `QuantizationTesterMixin` and `QuantizationCompileTesterMixin`

4. **Fix training gradient underflow** — [#13540](https://github.com/huggingface/diffusers/pull/13540)
   - Change autocast from `float16` to `bfloat16` to prevent gradient underflow in quantized training tests

5. **Reduce WanAnimate test input sizes** — [#13541](https://github.com/huggingface/diffusers/pull/13541)
   - Reduce spatial/temporal dimensions to prevent OOM on devices without FlashAttention (SDPA math backend materializes O(S²) attention matrix)

Each PR is independent and can be reviewed/merged separately. Will close this PR once the split PRs are up.

File: src/diffusers/quantizers/torchao/torchao_quantizer.py [hunk 1]
```diff
@@ -376,3 +376,17 @@ def is_trainable(self):
     @property
     def is_compileable(self) -> bool:
         return True
+
+    def _dequantize(self, model):
```
Comment (sayakpaul) [inline src/diffusers/quantizers/torchao/torchao_quantizer.py:380]: We shouldn't have dequantize here in this PR right?

File: tests/models/transformers/test_models_transformer_wan_animate.py [hunk 2]
```diff
@@ -224,7 +224,7 @@ def get_dummy_inputs(self):
         """Override to provide inputs matching the tiny Wan Animate model dimensions."""
         return {
             "hidden_states": randn_tensor(
-                (1, 36, 21, 64, 64), generator=self.generator, device=torch_device, dtype=self.torch_dtype
+                (1, 36, 5, 16, 16), generator=self.generator, device=torch_device, dtype=self.torch_dtype
```
Comment (sayakpaul) [inline tests/models/transformers/test_models_transformer_wan_animate.py:227]: Explain the changes.

File: tests/models/testing_utils/quantization.py [hunk 3]
```diff
@@ -1159,6 +1191,14 @@ class QuantizationCompileTesterMixin:
         - get_dummy_inputs(): Returns dict of inputs to pass to the model forward pass
     """
 
+    def _get_dummy_inputs_for_model(self, model):
+        inputs = self.get_dummy_inputs()
+        model_dtype = next(model.parameters()).dtype
+        return {
+            k: v.to(model_dtype) if isinstance(v, torch.Tensor) and v.is_floating_point() else v
+            for k, v in inputs.items()
+        }
```
Comment (sayakpaul) [inline tests/models/testing_utils/quantization.py:1200]: Why do we need to override?

File: src/diffusers/quantizers/torchao/torchao_quantizer.py (hunk 1 above)
Comment (jiqing-feng) [reply src/diffusers/quantizers/torchao/torchao_quantizer.py:380]: Yes, please review the change here: https://github.com/huggingface/diffusers/pull/13538

File: tests/models/transformers/test_models_transformer_wan_animate.py (hunk 2 above)
Comment (jiqing-feng) [reply tests/models/transformers/test_models_transformer_wan_animate.py:227]: It's for avoiding OOM, details see: https://github.com/huggingface/diffusers/pull/13541. Please let me know if you want comments in the code.

File: tests/models/testing_utils/quantization.py (hunk 3 above)
Comment (jiqing-feng) [reply tests/models/testing_utils/quantization.py:1200]: QuantizationCompileTesterMixin is an independent mixin that doesn't inherit from QuantizationTesterMixin. Test classes may use either one or both, so the method needs to be defined in both places.

Alternatively, I can extract it into a shared base class or a standalone utility function to avoid code duplication. Let me know which approach you prefer. Please review this change in https://github.com/huggingface/diffusers/pull/13539

**Comment by jiqing-feng** [comment]
Hi @sayakpaul . I have separated this PR into 5 small PRs, please review them 1 by 1 if it is easier for you. Thanks!

**Comment by jiqing-feng** [comment]
> Hi @sayakpaul . Thanks for the review! I've split this PR into 5 smaller independent PRs as suggested:
> 
> 1. **Enable TorchAO int4wo quantization tests on XPU** — [#13537](https://github.com/huggingface/diffusers/pull/13537)
>    
>    * Remove `_int4wo_skip` marker on XPU
>    * Add `int4_packing_format="plain_int32"` for XPU int4 quantization
> 2. **Implement `_dequantize` for TorchAO quantizer** — [#13538](https://github.com/huggingface/diffusers/pull/13538)
>    
>    * Add `_dequantize()` method to `TorchAoHfQuantizer`
>    * Fix `_verify_if_layer_quantized` to check `TorchAOBaseTensor` weight type
> 3. **Add `_get_dummy_inputs_for_model` helper** — [#13539](https://github.com/huggingface/diffusers/pull/13539)
>    
>    * Cast floating-point input tensors to model's parameter dtype automatically
>    * Defined in both `QuantizationTesterMixin` and `QuantizationCompileTesterMixin`
> 4. **Fix training gradient underflow** — [#13540](https://github.com/huggingface/diffusers/pull/13540)
>    
>    * Change autocast from `float16` to `bfloat16` to prevent gradient underflow in quantized training tests
> 5. **Reduce WanAnimate test input sizes** — [#13541](https://github.com/huggingface/diffusers/pull/13541)
>    
>    * Reduce spatial/temporal dimensions to prevent OOM on devices without FlashAttention (SDPA math backend materializes O(S²) attention matrix)
> 
> Each PR is independent and can be reviewed/merged separately. Will close this PR once the split PRs are up.

Hi @sayakpaul . Would you please review the 5 PRs I mentioned before? Thanks!
